src/components/NewWorkspaceModal.js

```diff
@@ -99,18 +99,25 @@ const NewWorkspaceModal = withDisplayName('NewWorkspaceModal', ({
       onSuccess(await Utils.cond(
         [cloneWorkspace, async () => {
           const workspace = await Ajax().Workspaces.workspace(cloneWorkspace.workspace.namespace, cloneWorkspace.workspace.name).clone(body)
+          if (isAzureBillingProject()) {
```

Comment:
Actually, could you move this createApp code out of here and do it in a separate wait (with its own error handling) after the workspace creation and event firing has happened? That way you don't have to duplicate the call in the 2 locations, and the error messaging could be more specific.

Reply:
Sorry I'm not quite sure I follow -- where exactly would you want to move the call?

Reply:
I am thinking that that the code in the `onSuccess` callback needs to look like this:

1) Instantiate the new workspace (through the conditional code that either clones an existing workspace or does does a create, plus its metrics event firing).

2) After the previous step has been executed, create the Leo app for Azure workspaces.

In other words, the code for creating the Leo app should only be in 1 place, instead of being inserted into both conditions in step 1.

As far as error handling, I'm not clear if the call to Leo needs to have its own error handling, or if things will be reasonable if we allow any errors to fall into the existing catch block. I recommend that you try doing something that will error (for example, just change the URL being hit to be wrong) and see what happens.

Reply:
I added some refactoring (creating the `createLeoApp` helper function. Let me know if this is alright with what you had in mind.

(Perhaps this is my lack of understanding with cloning workspaces), but aren't the two functions listed there exclusive of one another? E.g. I'm a bit stumped on the syntax you had in mind for only calling `createLeoApp` once

Reply:
I pushed up a change to the error handling. I want to make sure that the clearest possible thing happens if the Leo ajax call errors after the workspace creation already has succeeded. With what I pushed up, this is what you will see (and you can expand for details):

![image](https://user-images.githubusercontent.com/484484/213536643-1de7dd40-fe15-479a-8be4-15b121b4ce3c.png)

With what you had, you are stuck with the New Workspace modal left open and spinning, and at least when the error I mocked (just changing the URL), you didn't see the reason for the failure. 

Since the workspace itself has created and will be returned in the list of workspaces, we want to make sure that it is clear where the error occurred.

Note… you may wish a different error summary message than "Error starting Leo app".

Reply:
pretty sure the marketing term is "Workspace Services," so a good summary is "Error starting Workspace Services." You might wanna get Beth's input on that, though